plugin/src/main/java/io/jenkins/plugins/casc/impl/secrets/PropertiesSecretSource.java

```diff
@@ -32,30 +32,26 @@
      */
     public static final String SECRETS_DEFAULT_PATH = "/run/secrets/secrets.properties";
 
-    private Properties secrets;
+    private Properties secrets = new Properties();
 
     @Override
     public Optional<String> reveal(String secret) {
-        // lazy initialization
-        if (secrets == null) {
-            secrets = new Properties();
-            final String secretsEnv = System.getenv("SECRETS_FILE");
-            final String secretsPath = secretsEnv == null ? SECRETS_DEFAULT_PATH : secretsEnv;
-            final File secretsFile = new File(secretsPath);
-            if (secretsFile.exists() && secretsFile.isFile()) {
-                try (InputStream input = new FileInputStream(secretsFile)) {
-                    secrets.load(input);
-                } catch (IOException ioe) {
-                    LOGGER.log(Level.WARNING,
-                        "Source properties file " + secretsPath + " could not be loaded", ioe);
-                }
-            }
-        }
+        return Optional.ofNullable(secrets.getProperty(secret));
+    }
 
-        if (secrets.getProperty(secret) == null) {
-            return Optional.empty();
-        } else {
-            return Optional.of(secrets.getProperty(secret));
+    @Override
+    public void init() {
+        secrets.clear();
+        final String secretsEnv = System.getenv("SECRETS_FILE");
+        final String secretsPath = secretsEnv == null ? SECRETS_DEFAULT_PATH : secretsEnv;
+        final File secretsFile = new File(secretsPath);
+        if (secretsFile.exists() && secretsFile.isFile()) {
+            try (InputStream input = new FileInputStream(secretsFile)) {
+                secrets.load(input);
```

Comment:
Perhaps a good idea to move the `secrets.clear` closer to the `secrets.load`
```suggestion
        final String secretsEnv = System.getenv("SECRETS_FILE");
        final String secretsPath = secretsEnv == null ? SECRETS_DEFAULT_PATH : secretsEnv;
        final File secretsFile = new File(secretsPath);
        if (secretsFile.exists() && secretsFile.isFile()) {
            try (InputStream input = new FileInputStream(secretsFile)) {
                secrets.clear();
                secrets.load(input);
```